xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -548,6 +548,50 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
           m_mime = "video/dolby-vision";
           m_formatname = isDvhe ? "amc-dvhe" : "amc-dvh1";
           profile = 0; // not an HEVC profile
+
+          if (CJNIBase::GetSDKVersion() >= 24)
+          {
+            switch (m_hints.dovi.dv_profile)
+            {
+              case 0:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvavPer;
+                break;
+              case 1:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvavPen;
+                break;
+              case 2:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvheDer;
+                break;
+              case 3:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvheDen;
+                break;
+              case 4:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvheDtr;
+                break;
+              case 5:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvheStn;
+                break;
+              case 6:
+                profile = CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvheDth;
+                break;
+              case 7:
+                profile = 0; // DolbyVisionProfileDvheDtb
+                break;
+              case 8:
+                profile = CJNIBase::GetSDKVersion() >= 27
+                              ? CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvheSt
+                              : 0;
+                break;
+              case 9:
+                profile = CJNIBase::GetSDKVersion() >= 27
+                              ? CJNIMediaCodecInfoCodecProfileLevel::DolbyVisionProfileDvavSe
+                              : 0;
+                break;
+              default:
```

Comment:
default case seems redundant as profile is initially set to 0 and all for all cases NOT handled remains 0 anyway